src/components/structures/MessagePanel.tsx

```diff
@@ -467,6 +467,17 @@ export default class MessagePanel extends React.Component<IProps, IState> {
 
     // TODO: Implement granular (per-room) hide options
     public shouldShowEvent(mxEv: MatrixEvent, forceHideEvents = false): boolean {
+        if (this.props.hideThreadedMessages
+            && SettingsStore.getValue("feature_thread")) {
```

Comment:
Could fit on a single line, or if not I'd prefer `) {` on a new line, so it's easy to see where the block starts.